src/components/CartButton.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <div class="cart-button">
+    <action-button class="cart-button-btn" @click="onViewCart"
+      ><i class="cart-button-btn-icon"></i
+    ></action-button>
+    <div class="cart-button-count" v-if="countItems > 0">{{ countItems }}</div>
+  </div>
+</template>
+
+<script setup>
+import ActionButton from "@/components/general/ActionButton.vue";
+import { useRouter } from "vue-router";
+import { createNamespacedHelpers } from "vuex-composition-helpers";
+import { CART_COUNT } from "@/store/storeConstants";
+
+const { useGetters } = createNamespacedHelpers("cart");
```

Comment:
dont use store (actions or getters) on low-level components
must be outside-props or callback or emit-action